More robust validation of avatar texture settings

BoatFloater opened the pull request.

BoatFloater committed:
only validate avatar streaming mip maps if mip maps are enabled

Toys0125 commented:
In my opinion, needs to be a more advanced check due to some people will forget to enable mipmaping for their avatar textures. And if you enable the streaming even though mipmap is off will pass the check currently and wont affect the texture since its off. So might force enable streaming textures for all textures will be better.

BoatFloater commented:
Mip maps are enabled by default; if they are disabled that means the person who imported them specifically turned them off.
The check is not relevant to textures with disabled mip maps. I think it's better for this check to be accurate than to use it as a nudge to encourage users to enable mip maps on various textures.
Some textures, for example shader LUTs, might come with mip maps off and could confuse the user why they are getting a warning about it.

BoatFloater commented:
Sure, but I don't understand why there should be a warning instructing the user to enable mip streaming on a texture where 'Generate Mipmap' is not checked
<img width="562" height="201" alt="image" src="https://github.com/user-attachments/assets/dc25f2a4-caed-4db0-ac2b-ac4af82f747d" />

BoatFloater commented:
But the check does not validate or enforce 'Generate Mipmap'. Only 'Stream Mipmap Levels', which does nothing if 'Generate Mipmap' is off.

BoatFloater committed:
more robust texture checks and validator improvements

BoatFloater commented:
I made the checks more robust:
* check for 'Generate Mipmap' on textures >256
* check for 'Stream Mipmap Levels' on all textures which have 'Generate Mipmap' enabled
* Added Auto-Fix actions for both checks and the existing check for >4k textures

I also noticed some issues with the auto-fix buttons (they were gray before the first hover and the text overflowed) so I fixed those and added a warning color and optional action title; let me know if that would be better resolved another way.
